### PR TITLE
Endpoints urls with slash

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- revert stripping of trailing slashes for endpoint urls and replace it with a debug message
- fixes https://github.com/RasaHQ/rasa/issues/3776#event-2412592607

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/endpoints.py [hunk 1]
```diff
@@ -46,7 +46,13 @@ def concat_url(base: Text, subpath: Optional[Text]) -> Text:
...
+        if base.endswith("/"):
+            logger.debug(
+                "The URL '{}' has a trailing slash. Please make sure the "
+                "target server supports trailing slashes for this "
+                "endpoint.".format(base)
+            )
+        return base
 
     url = base
     if not base.endswith("/"):
```
Comment (erohmensing) [inline rasa/utils/endpoints.py]: I'm not exactly sure what's going on here, should the debug message also go in this if statement?

Comment (wochinge) [reply rasa/utils/endpoints.py]: I decided against, cause if `subpath` values are added by Rasa code (and not externally) and we should know what we are doing and if we add a slash or not.

Comment (erohmensing) [reply rasa/utils/endpoints.py]: Ah okay I got it, I was confused bc i was missing that these were 2 cases with and w/o subpath

**Comment by Wing-e7** [comment]
Still doesnt solve the 404 issue. Please help

**Comment by erohmensing** [comment]
@Wing-e7 please open a bug report with full information and ideally reproducible behavior if you think there is still a bug!

**Pull request merged**